processing-status-notifications-fa/src/main/kotlin/gov/cdc/ocio/cache/InMemoryCache.kt

```diff
@@ -0,0 +1,106 @@
+package gov.cdc.ocio.cache
+
+import gov.cdc.ocio.exceptions.BadStateException
+import gov.cdc.ocio.model.cache.NotificationSubscriber
+import gov.cdc.ocio.model.http.SubscriptionType
+import mu.KotlinLogging
+import java.util.*
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.collections.HashMap
+
+
+object InMemoryCache {
+    private val logger = KotlinLogging.logger {}
+    private val readWriteLock = ReentrantReadWriteLock()
+
+    // Cache to store "SubscriptionRule HashSet ->  SubscriptionId"
+    private val subscriptionRuleCache = HashMap<String, String>()
+
+    // Cache to store "SubscriptionId ->  Subscriber Info (Email or Url and type of subscription)"
+    private val subscriberCache = HashMap<String, NotificationSubscriber>()
+
+
+    fun updateCacheForSubscription(subscriptionRule: String,
```

Comment:
Please add method headers to all methods, with descriptions of what the method does, the params, return value and exceptions thrown.  Please see the main PS API FA for examples.